src/Cli/Microsoft.DotNet.Cli.Sln.Internal/SlnFile.cs

```diff
@@ -278,6 +278,11 @@ public string FilePath
             }
         }
 
+        public bool AreBuildConfigurationsApplicable()
+        {
+            return Path.GetExtension(_filePath) == ".shproj";
```

Comment:
Should this comparison ignore case?

Reply:
This returns false for ".csproj" and true for ".shproj" but should do the opposite.

Reply:
Based on the comment in SlnFileExtensions.cs

<https://github.com/dotnet/sdk/blob/8f9105dec7ab2e7468b5ab1828674285548e5ed0/src/Cli/dotnet/SlnFileExtensions.cs#L76-L77>

perhaps it would be better to make AreBuildConfigurationsApplicable an extension method in SlnFileExtensions, as it does not seem lower-level than AddDefaultBuildConfigurations.

Reply:
Fixed both of your comments. Earlier I named the method different and haven't adjusted return expression